app/javascript/src/components/Dashboard/Notes/index.jsx

```diff
@@ -1,102 +1,60 @@
-import React, { useState, useEffect } from "react";
+import React, { useState } from "react";
 
 import EmptyNotesListImage from "images/EmptyNotesList";
-import { Delete } from "neetoicons";
-import { Button, PageLoader } from "neetoui";
-import { Container, Header, SubHeader } from "neetoui/layouts";
+import { Button } from "neetoui";
+import { Container, Header } from "neetoui/layouts";
 
-import notesApi from "apis/notes";
 import EmptyState from "components/Common/EmptyState";
+import Menubar from "components/Common/Menubar";
 
-import DeleteAlert from "./DeleteAlert";
-import NoteTable from "./NoteTable";
+import Card from "./Card";
+import { NOTES_DATA } from "./constants";
 import NewNotePane from "./Pane/CreateNote";
 
 const Notes = () => {
-  const [loading, setLoading] = useState(true);
-  const [showNewNotePane, setShowNewNotePane] = useState(false);
-  const [showDeleteAlert, setShowDeleteAlert] = useState(false);
   const [searchTerm, setSearchTerm] = useState("");
-  const [selectedNoteIds, setSelectedNoteIds] = useState([]);
-  const [notes, setNotes] = useState([]);
-
-  useEffect(() => {
-    fetchNotes();
-  }, []);
-
-  const fetchNotes = async () => {
-    try {
-      setLoading(true);
-      const { data } = await notesApi.fetch();
-      setNotes(data.notes);
-    } catch (error) {
-      logger.error(error);
-    } finally {
-      setLoading(false);
-    }
-  };
-
-  if (loading) {
-    return <PageLoader />;
-  }
+  const [showMenu, setShowMenu] = useState(true);
+  const [showNewNotePane, setShowNewNotePane] = useState(false);
 
   return (
-    <Container>
-      <Header
-        title="Notes"
-        actionBlock={
-          <Button
-            onClick={() => setShowNewNotePane(true)}
-            label="Add New Note"
-            icon="ri-add-line"
-          />
-        }
-        searchProps={{
-          value: searchTerm,
-          onChange: e => setSearchTerm(e.target.value),
-        }}
-      />
-      {notes.length ? (
-        <>
-          <SubHeader
-            rightActionBlock={
-              <Button
-                label="Delete"
-                icon={Delete}
-                onClick={() => setShowDeleteAlert(true)}
-                disabled={!selectedNoteIds.length}
-              />
-            }
-          />
-          <NoteTable
-            setSelectedNoteIds={setSelectedNoteIds}
-            notes={notes}
-            fetchNotes={fetchNotes}
-          />
-        </>
-      ) : (
-        <EmptyState
-          image={EmptyNotesListImage}
-          title="Looks like you don't have any notes!"
-          subtitle="Add your notes to send customized emails to them."
-          primaryAction={() => setShowNewNotePane(true)}
-          primaryActionLabel="Add New Note"
+    <>
+      <Menubar showMenu={showMenu} title="Notes" />
+      <Container>
+        <Header
+          title="All Notes"
+          menuBarToggle={() => setShowMenu(prev => !prev)}
```

Comment:
What's your reasoning for using prevState from within the setter over here rather than using `!showMenu`? What are the advantages of using this pattern over here?

Reply:
I'm little confused here on what the best approach is to toggle state. Ive read somewhere that its best to use prevState  when toggling because the next state is dependent on the prevState. Can you please tell me what approach should i follow ?

Reply:
@vijay2136 

Go through the below mentioned articles and try to understand why we should be using previous state callback over here:
- https://stackoverflow.com/questions/58193166/usestate-hook-setter-incorrectly-overwrites-state
- https://medium.com/nerd-for-tech/react-hooks-with-closures-usestate-v-s-usereducer-9e0c20e81051
- https://dev.to/anpos231/react-hooks-the-closure-hell-71m

Once you go through them and understand what's going on, then please do type in a reply for this comment pointing in simple terms why we are preferring prevState over here. This is to ensure whether you've learn the right thing or not.

Reply:
when you have a function defined inside of another function, that inner function has access to the variables and scope of the outer function, even if the outer function finishes executing and those variables are no longer accessible outside of that function. So its better to use prevState callback to avoid the closure problem.

Reply:
@vijay2136 

That's just a part of it.

Main issue is this: https://stackoverflow.com/a/58193404

Reply:
Yes, I was trying to say that in simpler terms.